src/shared/language-util.ts

```diff
@@ -104,100 +105,82 @@ export function insertBefore (grammar: Grammar, before: string, insert: GrammarT
  *     'color': /\b(?:red|green|blue)\b/
  * });
  */
-export function extend (grammar: Grammar, id: string, reDef: Grammar): Grammar {
-	const lang = cloneGrammar(grammar, id);
+export function extend (base: Grammar, id: string, grammar: Grammar): Grammar {
```

Comment:
We can probably ditch `id` here (it's not used anywhere in the function body). `deepClone()` doesn't need it as a parameter anymore.